p2p/src/peer_manager/mod.rs

```diff
@@ -83,7 +91,13 @@ where
     /// Peer database
     peerdb: peerdb::PeerDb<T>,
 
+    /// Last time when heartbeat was called
     last_heartbeat: Instant,
+
+    /// All addresses that were announced to or from some peer.
+    /// Used to prevent infinity loops while broadcasting addresses.
+    /// Bitcoin Core uses bloom filter for that.
```

Comment:
Why don't we do that?

Reply:
Let's please do this later. I added TODO.